server/src/main/java/org/opensearch/search/aggregations/bucket/FilterRewriteHelper.java

```diff
@@ -252,8 +304,11 @@ static boolean tryFastFilterAggregation(
                     ),
                     counts[i]
                 );
+                s++;
+                if (s > size) return true;
```

Comment:
Use the size information while generating buckets instead of here

Reply:
https://github.com/opensearch-project/OpenSearch/pull/11505#discussion_r1440821864